### ISSUE TITLE
Issue with generated code: database.g.dart file

### ISSUE THREAD

**Issue by chani22** [simolus3/moor#3043]
Hello,

I am encountering an issue with the generated code from Drift. When I run the code generation using `build_runner`, I receive the following error by my ide(vscode):
```
The argument type 'Object?' can't be assigned to the parameter type 'Expression<int>'.
```

If I ignore the above error and run flutter run, I encounter the following build-time error.
```
Error (Xcode): lib/data/database/database.g.dart:7976:25: Error: The argument type 'Object?' can't be assigned
to the parameter type 'Expression<int>'.
```

Below is the screen of error message reported by VSCode.
<img width="764" alt="스크린샷 2024-06-11 오후 5 59 47" src="https://github.com/simolus3/drift/assets/88638570/6773744f-1665-415d-9801-ec220f299121">

Here is a part of the `database.dart` code I wrote, defining the BuildingTable.
```
class BuildingTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable()();
  IntColumn get gid32 => integer().nullable()();

  TextColumn get buildingName => text().withLength(max: 50)();
  TextColumn get buildingIcon => text().withLength(max: 50)();
  IntColumn get iconColorValue => integer()();
  TextColumn get address => text().withLength(max: 100).nullable()();
  TextColumn get buildingType => text().withLength(max: 70).nullable()();
  TextColumn get housingType => text().withLength(max: 70).nullable()();

  IntColumn get purchaseDate => integer().nullable()();
  IntColumn get purchasePrice => integer().nullable()();
  IntColumn get marketPrice => integer().nullable()();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();
}
```

Below is a part of the yaml file defining the related dependencies.
```
  drift: ^2.18.0
  sqlite3_flutter_libs: ^0.5.23
  path_provider: ^2.1.3
  path: ^1.9.0
  watcher: ^1.1.0

dev_dependencies:
  flutter_test:
    sdk: flutter
  drift_dev: ^2.18.0
  build_runner: ^2.4.11
```

How can I resolve this issue?

**Comment by simolus3** [comment]
Thanks for the report! Unfortunately I can't reproduce this locally by pasting the table code alone (it generates the same code but I don't get the error about `column` having an invalid type). I don't think this is an analyzer problem either since the same thing is happening when compiling the app.
If you hover over a `composableBuilder` invocation in VS Code, what does the popup say its type is? It should be something like `ColumnFilters<String> Function({required ColumnFilters<String> Function(GeneratedColumn<String>, Set<JoinBuilder>) builder, required GeneratedColumn<String> column})`, but I suspect there may be something going wrong here.

As an interim workaround, you can disable generating the code causing this error by creating [builder options](https://drift.simonbinder.eu/docs/advanced-features/builder_options/) with this content:

```yaml
targets:
  $default:
    builders:
      drift_dev:
        options:
          generate_manager: false
```

But obviously we want to know what's happening here so that you can keep that code. also cc @dickermoshe (not urgent though).

**Comment by dickermoshe** [comment]
I’ll take a look, but if it not reproducible, there isn’t much I can do…

**Comment by chani22** [comment]
@simolus3 
The temporary workaround you provided works effectively. Thank you very much.

And, I agree that we need to find a fundamental solution.

I have attached the popup that appears when hovering over `composableBuilder` via vscode.
<img width="757" alt="스크린샷 2024-06-12 오전 12 20 22" src="https://github.com/simolus3/drift/assets/88638570/1dca7f03-bd67-4561-8055-551514cfa85a">

I am ready to provide any additional information needed to resolve this issue. Please let me know if there is anything else you need.

**Comment by dickermoshe** [comment]
@chani22 I need the database class too

**Comment by dickermoshe** [comment]
You've only posted the table class

**Comment by chani22** [comment]
@dickermoshe 

Here is my full `database.dart`file code.
(Would it be possible to mask some parts of the code below once this issue is resolved? This is our core business, so we are cautious about exposing it externally.)
```
// These imports are only needed to open the database
import 'package:drift/native.dart';
import 'package:flutter_application_1/bridge/PrefsService.dart';
import 'package:flutter_application_1/bridge/SynchronizationService.dart';
import 'package:flutter_application_1/data/entities/DataManager.dart';
import 'package:flutter_application_1/main.dart';
import 'package:flutter_local_notifications/flutter_local_notifications.dart';
import 'package:path_provider/path_provider.dart';
import 'package:path/path.dart' as p;
import 'package:drift/drift.dart';
import 'dart:io';

// assuming that your file is called filename.dart. This will give an error at first,
// but it's needed for drift to know about the generated code
part 'database.g.dart';

class BuildingTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get gid32 => integer().nullable().withDefault(const Constant(null))();

  TextColumn get buildingName => text().withLength(max: 50)();
  TextColumn get buildingIcon => text().withLength(max: 50)();
  IntColumn get iconColorValue => integer()();
  TextColumn get address => text().withLength(max: 100).nullable()();
  TextColumn get buildingType => text().withLength(max: 70).nullable()();
  TextColumn get housingType => text().withLength(max: 70).nullable()();

  IntColumn get purchaseDate => integer().nullable()();
  IntColumn get purchasePrice => integer().nullable()();
  IntColumn get marketPrice => integer().nullable()();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();

  @override
  Set<Column> get primaryKey => {id, gid0, gid32};
}

class UnitTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get gid32 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get buildingId => integer()();
  IntColumn get buildingGid0 => integer().nullable()();
  IntColumn get buildingGid32 => integer().nullable()();

  TextColumn get bNum => text().withLength(max: 32)();
  TextColumn get uNum => text().withLength(max: 32)();

  IntColumn get area => integer().nullable()();
  TextColumn get direction => text().withLength(max: 100)();
  TextColumn get structure => text().withLength(max: 100)();
  //TextColumn get description => text().nullable().withLength(min: 0, max: 500)();

  IntColumn get purchaseDate => integer().nullable()();
  IntColumn get purchasePrice => integer().nullable()();
  IntColumn get marketPrice => integer().nullable()();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();

  @override
  Set<Column> get primaryKey => {id, gid0, gid32};
}

class ContractTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get gid32 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get buildingId => integer()();
  IntColumn get buildingGid0 => integer().nullable()();
  IntColumn get buildingGid32 => integer().nullable()();
  IntColumn get unitId => integer()();
  IntColumn get unitGid0 => integer().nullable()();
  IntColumn get unitGid32 => integer().nullable()();

  IntColumn get prevContractId => integer().nullable()();
  IntColumn get prevContractGid0 => integer().nullable()();
  IntColumn get prevContractGid32 => integer().nullable()();
  IntColumn get nextContractId => integer().nullable()();
  IntColumn get nextContractGid0 => integer().nullable()();
  IntColumn get nextContractGid32 => integer().nullable()();

  IntColumn get contractDate => integer()();
  IntColumn get expirationDate => integer()();
  IntColumn get midTerminationDate => integer().nullable()();

  TextColumn get rentType => text().withLength(max: 70)();

  IntColumn get deposit => integer()();
  IntColumn get rentalFee => integer().nullable()();
  IntColumn get maintenanceFee => integer().nullable()();
  TextColumn get rentalFeePaymentType => text().nullable().withLength(max: 70)();

  IntColumn get fullyAdvancePaidRentalFee => integer().nullable()();
  IntColumn get fullyAdvancePaidMaintenanceFee => integer().nullable()();

  TextColumn get tenantName => text().nullable().withLength(max: 100)();
  TextColumn get tenantPhone => text().nullable().withLength(max: 100)();
  TextColumn get agencyName => text().nullable().withLength(max: 100)();
  TextColumn get agencyPhone => text().nullable().withLength(max: 100)();
  IntColumn get brokerageFee => integer().nullable()();
  IntColumn get brokerageFeeAccountId => integer().nullable()();
  IntColumn get brokerageFeeAccountGid0 => integer().nullable()();
  IntColumn get brokerageFeeAccountGid32 => integer().nullable()();
  IntColumn get depositReceiveDate => integer().nullable().withDefault(const Constant(null))();
  IntColumn get fullyAdvancePaidRentalFeeReceiveDate => integer().nullable().withDefault(const Constant(null))();
  TextColumn get rentalFeeReceiveMiDate => text().nullable().withDefault(const Constant(null))();
  TextColumn get maintenanceFeeReceiveMiDate => text().nullable().withDefault(const Constant(null))();
  //TextColumn get description => text().nullable().withLength(min: 0, max: 500)();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();

  @override
  Set<Column> get primaryKey => {id, gid0, gid32};
}

class BillTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get gid32 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get buildingId => integer()();
  IntColumn get buildingGid0 => integer().nullable()();
  IntColumn get buildingGid32 => integer().nullable()();
  IntColumn get unitId => integer()();
  IntColumn get unitGid0 => integer().nullable()();
  IntColumn get unitGid32 => integer().nullable()();
  IntColumn get contractId => integer()();
  IntColumn get contractGid0 => integer().nullable()();
  IntColumn get contractGid32 => integer().nullable()();

  IntColumn get depositBillId => integer().nullable()();
  IntColumn get depositBillGid0 => integer().nullable()();
  IntColumn get depositBillGid32 => integer().nullable()();

  TextColumn get billType => text().withLength(max: 70)();
  TextColumn get paymentStatus => text().withLength(max: 70)();
  TextColumn get prevPaymentStatus => text().withLength(max: 70).nullable()();

  // 해당 월
  IntColumn get periodMonth => integer()();

  // // 기간 시작/끝
  IntColumn get periodStartDate => integer().nullable()();
  IntColumn get periodEndDate => integer().nullable()();

  // // 납입일
  IntColumn get paymentDate => integer().nullable()();
  IntColumn get amountReceivedDate => integer().nullable()();

  IntColumn get willReturnDate => integer().nullable()();
  IntColumn get realReturnDate => integer().nullable()();

  // 금액
  IntColumn get deposit => integer().nullable()();
  IntColumn get remainDeposit => integer().nullable()();
  IntColumn get rentalFee => integer().nullable()();
  IntColumn get maintenanceFee => integer().nullable()();
  IntColumn get fullyAdvancePaidRentalFee => integer().nullable()();
  IntColumn get fullyAdvancePaidMaintenanceFee => integer().nullable()();

  IntColumn get amountReceived => integer().withDefault(const Constant(0))();

  BoolColumn get tableDisplay => boolean().withDefault(const Constant(true))();
  BoolColumn get listDisplay => boolean().withDefault(const Constant(true))();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();

  @override
  Set<Column> get primaryKey => {id, gid0, gid32};
}

class AccountTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get gid32 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get buildingId => integer().nullable()();
  IntColumn get buildingGid0 => integer().nullable()();
  IntColumn get buildingGid32 => integer().nullable()();
  IntColumn get unitId => integer().nullable()();
  IntColumn get unitGid0 => integer().nullable()();
  IntColumn get unitGid32 => integer().nullable()();
  IntColumn get contractId => integer().nullable()();
  IntColumn get contractGid0 => integer().nullable()();
  IntColumn get contractGid32 => integer().nullable()();

  TextColumn get accountType => text().withLength(max: 70)();
  IntColumn get amount => integer()();
  TextColumn get content => text().withLength(max: 100)();

  IntColumn get incurredDate => integer()();
  BoolColumn get editable => boolean()();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();

  @override
  Set<Column> get primaryKey => {id, gid0, gid32};
}

class MemoTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get gid32 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get buildingId => integer().nullable()();
  IntColumn get buildingGid0 => integer().nullable()();
  IntColumn get buildingGid32 => integer().nullable()();
  IntColumn get unitId => integer().nullable()();
  IntColumn get unitGid0 => integer().nullable()();
  IntColumn get unitGid32 => integer().nullable()();
  IntColumn get contractId => integer().nullable()();
  IntColumn get contractGid0 => integer().nullable()();
  IntColumn get contractGid32 => integer().nullable()();
  IntColumn get billId => integer().nullable()();
  IntColumn get billGid0 => integer().nullable()();
  IntColumn get billGid32 => integer().nullable()();

  TextColumn get content => text().withLength(max: 500)();
  BoolColumn get isImportant => boolean()();

  DateTimeColumn get lastModifiedDateTime => dateTime()();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();

  @override
  Set<Column> get primaryKey => {id, gid0, gid32};
}

class NotificationTable extends Table {
  IntColumn get id => integer()();
  IntColumn get gid0 => integer().nullable().withDefault(const Constant(null))();
  IntColumn get gid32 => integer().nullable().withDefault(const Constant(null))();

  TextColumn get notificationType => text().withLength(min: 0, max: 70)();
  TextColumn get data => text()();

  DateTimeColumn get createdDateTime => dateTime()();
  DateTimeColumn get lastUpdatedDateTime => dateTime()();
  BoolColumn get isRead => boolean().withDefault(const Constant(false))();
  BoolColumn get isDeleted => boolean().withDefault(const Constant(false))();
}

class NotificationScheduleTable extends Table {
  IntColumn get id => integer()();
  IntColumn get notificationId => integer().nullable()();

  TextColumn get notificationType => text().withLength(min: 0, max: 70)();
  TextColumn get data => text()();

  DateTimeColumn get triggerDateTime => dateTime()();
  BoolColumn get isTriggered => boolean().withDefault(const Constant(false))();
}

LazyDatabase _openConnection() {
  // the LazyDatabase util lets us find the right location for the file async.
  return LazyDatabase(() async {
    // put the database file, called db.sqlite here, into the documents folder
    // for your app.
    final dbFolder = await getApplicationDocumentsDirectory();
    final file = File(p.join(dbFolder.path, 'db.sqlite'));
    return NativeDatabase(file);
  });
}

// this annotation tells drift to prepare a database class that uses both of the
// tables we just defined. We'll see how to use that database class in a moment.
@DriftDatabase(tables: [BuildingTable, UnitTable, ContractTable, BillTable, AccountTable, MemoTable, NotificationTable, NotificationScheduleTable] )
class MyDatabase extends _$MyDatabase{

  MyDatabase() : super(_openConnection());

  @override
  //int get schemaVersion => 33;
  //int get schemaVersion => 34; // NotificationTable
  //int get schemaVersion => 35; // NotificationScheduleTable
  //int get schemaVersion => 36; // add column at contract (depositReceiveDate, rentalFeeReceiveMiDate, maintenanceFeeReceiveMiDate)
  //int get schemaVersion => 37; // add column at contract (fullyAdvancePaidRentalFeeReceiveDate)
  
  // remove column at building (category, loan) add column at building(buildingType, housingType, purchaseDate), add column at unit (purchaseDate, purchasePrice, marketPrice)
  int get schemaVersion => 38;

  @override
  MigrationStrategy get migration => MigrationStrategy(
    onCreate: (Migrator m) async {
      await m.createAll(); // create all tables

      //await setUpInitalData();

    },
    onUpgrade: (Migrator m, int from, int to) async {

      await transaction(() async {
        if (from < 34) {
          await m.create(notificationTable);
        }

        if (from < 35) {
          await m.create(notificationScheduleTable);
        }

        if (from < 36) {
          await m.addColumn(contractTable, contractTable.depositReceiveDate);
          await m.addColumn(contractTable, contractTable.rentalFeeReceiveMiDate);
          await m.addColumn(contractTable, contractTable.maintenanceFeeReceiveMiDate);
        }

        if (from < 37) {
          await m.addColumn(contractTable, contractTable.fullyAdvancePaidRentalFeeReceiveDate);
        }

        if (from < 38) {
          await m.addColumn(buildingTable, buildingTable.buildingType);
          await m.addColumn(buildingTable, buildingTable.housingType);
          await m.addColumn(buildingTable, buildingTable.purchaseDate);

          await m.addColumn(unitTable, unitTable.purchaseDate);
          await m.addColumn(unitTable, unitTable.purchasePrice);
          await m.addColumn(unitTable, unitTable.marketPrice);
        }
      });
    },
  );

  Future resetDb() async {
    Migrator m = Migrator(database!);

    DataManager.closeStream();

    await notificationScheduleDao.removeAll();
    await notificationDao.removeAll();
    await memoDao.removeAll();
    await accountDao.removeAll();
    await billDao.removeAll();
    await contractDao.removeAll();
    await unitDao.removeAll();
    await buildingDao.removeAll();
    
    await removeAllData(m);
    await m.createAll();
    await PrefsService().setNewNotification(false);
    await PrefsService().setNotificationScheduleBounaryDate(null);
    await FlutterLocalNotificationsPlugin().cancelAll();
    DataManager.bindStream();
    SynchronizationService().updateUsageStatus();
    print("reset database complete");
  }

  Future removeAllData(Migrator m) async {
    await m.drop(notificationScheduleTable);
    await m.drop(notificationTable);
    await m.drop(memoTable);
    await m.drop(accountTable);
    await m.drop(billTable);
    await m.drop(contractTable);
    await m.drop(unitTable);
    await m.drop(buildingTable);
    // await rentalFeeHistoryDao.removeAll();
    // await contractDao.removeAll();
    // await unitDao.removeAll();
    // await buildingDao.removeAll();
  }
}
```

**Comment by dickermoshe** [comment]
I didn't need the full file.
You can delete it now.

**Comment by dickermoshe** [comment]
@chani22 
This is still working fine by me.

Do you have any `build.yaml` file that has some other settings?
What's your output of `flutter doctor`?
Did you try building with `dart run build_runner build --delete-conflicting-outputs`?
What about a `flutter clean`?

**Comment by chani22** [comment]
@dickermoshe 
There was no build.yaml file before simolus3 provided the temporary workaround.

This is output of `flutter doctor`
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.1, on macOS 14.5 23F79 darwin-arm64, locale ko-KR)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.3)
[✓] VS Code (version 1.81.1)
[✓] Connected device (5 available)
[✓] Network resources

• No issues found!
```

I have tried both `dart run build_runner build --delete-conflicting-outputs` and `flutter clean` multiple times, and I just tried both methods again, but I still confirmed that the error code exists in the database.g.dart file.

**Comment by dickermoshe** [comment]
What about `dart info`?

**Comment by chani22** [comment]
Here is output of `dart info`

```

If providing this information as part of reporting a bug, please review the information
below to ensure it only contains things you're comfortable posting publicly.

#### General info

- Dart 3.4.1 (stable) (Tue May 21 15:46:25 2024 +0000) on "macos_arm64"
- on macos / Version 14.5 (Build 23F79)
- locale is ko-KR

#### Project info

- sdk constraint: '>=2.17.0 <=3.4.1'
- dependencies: animated_check, awesome_notifications, badges, change_app_package_name, cloud_firestore, conditional_parent_widget, convex_bottom_bar, cupertino_icons, data_table_2, device_info_plus, dotted_border, drift, dropdown_button2, firebase_analytics, firebase_core, firebase_crashlytics, firebase_messaging, firebase_remote_config, flutter, flutter_bounceable, flutter_cupertino_date_picker_fork, flutter_easyloading, flutter_inappwebview, flutter_inset_box_shadow, flutter_local_notifications, flutter_material_pickers, flutter_native_splash, flutter_screen_lock, flutter_sticky_header, flutter_svg, flutter_swiper_null_safety, flutter_toggle_tab, flutter_typeahead, fluttertoast, font_awesome_flutter, get, getwidget, glassmorphism, group_button, hexcolor, http, internet_connection_checker, intl, is_first_run, jwt_decode, kakao_flutter_sdk, kpostal, linked_scroll_controller, logger, lottie, modal_side_sheet, multi_select_flutter, mutex, notification_permissions, package_info_plus, path, path_provider, percent_indicator, rect_getter, shared_preferences, shimmer, show_up_animation, sign_in_with_apple, smooth_page_indicator, sqlite3_flutter_libs, sticky_headers, store_redirect, styled_widget, substring_highlight, table_calendar, transparent_pointer, url_launcher, watcher, webview_flutter, word_break_text
- dev_dependencies: build_runner, drift_dev, flutter_launcher_icons, flutter_lints, flutter_oss_licenses, flutter_test
- elided dependencies: 1

#### Process info

| Memory |  CPU | Elapsed time | Command line                                                                    |
| -----: | ---: | -----------: | ------------------------------------------------------------------------------- |
|   1 MB | 0.0% |  04-01:43:48 | dart devtools --machine --allow-embedding --dtd-uri ws:<path>/fZvSUgQ0B7I3fiaD  |
|   1 MB | 0.0% |  04-01:24:19 | dart devtools --machine --allow-embedding --dtd-uri ws:<path>/pB0ZQGL0dkj0ctw3  |
|   6 MB | 0.0% |  04-01:43:48 | dart language-server --protocol=lsp --client-id=VS-Code --client-version=3.90.0 |
|   6 MB | 0.0% |  04-01:24:19 | dart language-server --protocol=lsp --client-id=VS-Code --client-version=3.90.0 |
|   1 MB | 0.0% |  04-01:43:48 | dart tooling-daemon --machine                                                   |
|   1 MB | 0.0% |  04-01:24:19 | dart tooling-daemon --machine                                                   |
|   4 MB | 0.0% |  04-01:24:19 | flutter_tools.snapshot daemon   
```

**Comment by dickermoshe** [comment]
Hmmm, this is really strange.
I've got the same with no issues.

What does hovering over `$state.table.id` reveal?
By me:
![image](https://github.com/simolus3/drift/assets/75931499/ffc780ba-8f1b-44ec-9dd7-bbd5c98b3e09)

**Comment by dickermoshe** [comment]
Can you post the generated code?
You can email it to me privately if you feel more comfortable
dickermoshe@gmail.com

**Comment by chani22** [comment]
Same type hovering over on `$state.table.id`

<img width="614" alt="스크린샷 2024-06-12 오전 1 11 49" src="https://github.com/simolus3/drift/assets/88638570/6804d0cd-8953-49be-9a35-0adc85012ec5">

**Comment by chani22** [comment]
@dickermoshe 
I just sent the full code of the database.g.dart file to the email you provided. Thank you for your consideration.

**Comment by dickermoshe** [comment]
That is a fairly nasty bug.

The generics on `composableBuilder` state that the column provided is passed into the builder.
<img width="389" alt="image" src="https://github.com/simolus3/drift/assets/75931499/1ed4fdf7-5793-4f73-8bbf-c1a104b2f136">
<img width="1156" alt="image" src="https://github.com/simolus3/drift/assets/75931499/6ecd9255-a27e-4f7c-ad54-456a5f3a00ab">

@simolus3 
This seems like a dart bug.

**Comment by dickermoshe** [comment]
@chani22 
What about hovering over `column` in `builder: (column, joinBuilders) =>`?

**Comment by dickermoshe** [comment]
@chani22 Your generated code matches mine.

**Comment by chani22** [comment]
Here is what appears when hovering over the `column`.

<img width="502" alt="스크린샷 2024-06-12 오전 1 28 34" src="https://github.com/simolus3/drift/assets/88638570/fa4722f3-cd99-4e3f-8d45-dfc305d024e0">

**Comment by dickermoshe** [comment]
This is really odd.
If this a bug in dart, the next steps will be long and painful.
I want to make sure that we aren't the problem.
I'll get back to you

**Comment by chani22** [comment]
I sincerely appreciate your prompt response. I will proceed with the development using the temporary workaround you provided. We will address the issue again once the bug is resolved.

Thank you once again.

**Comment by simolus3** [comment]
Just to be sure - this also prevents the code from running? I've sometimes seen issues where the analyzer reports phantom type errors but the front end gets it right. That's definitely an analyzer bug, but if the compiler also refuses to run the code with the same reason, chances are we're misusing generics in a subtle way (but then it's weird that it only fails here).

One possible approach to diagnose this further could be to make the generated code more verbose (it could spell out the generics in the `composableBuilder` invocation or add explicit types to the function expression passed to `builder`).

**Comment by dickermoshe** [comment]
@chani22 
Try replacing this on `BuildingTableTableFilterComposer`
```dart
ColumnFilters<int> get id => $state.composableBuilder(
      column: $state.table.id,
      builder: (column, joinBuilders) =>
          ColumnFilters(column, joinBuilders: joinBuilders));
```
with this:
```dart
ColumnFilters<int> get id => $state.composableBuilder<ColumnFilters<int>,GeneratedColumn<int>>(
      column: $state.table.id,
      builder: (column, joinBuilders) =>
          ColumnFilters(column, joinBuilders: joinBuilders));
```

I've added the generics explicitly, maybe that will help?

**Comment by dickermoshe** [comment]
@simolus3 
Yeah, it breaks on flutter run as stated in the initial bug report

> If I ignore the above error and run flutter run, I encounter the following build-time error.
> 
> Error (Xcode): lib/data/database/database.g.dart:7976:25: Error: The argument type 'Object?' can't be assigned
> to the parameter type 'Expression<int>'.

**Comment by lance-mlabs** [comment]
Hi all, Im also running into this same issue with generated `database.g.dart`. Hope the priority of this issue can be escalated for swift resolution. I will also resort to the above temporary workaround in the meantime.

**Comment by dickermoshe** [comment]
@lance-mlabs 
We are trying to find a way to reproduce this.
Until then there is nothing we can do to help.

I'm  assuming there is another import/code thats polluting the namespace.

Can you try removing all the code from `database.dart` that is not the Tables and Database and re-generate.

```dart
// Tables
class TableName extends Table {
  //...
}
// Database
@DriftDatabase(tables: [TableName ])
class Database extends _$Database {}
```

Also remove all imports that aren't from drift.

Are you still getting an error?

**Comment by simolus3** [comment]
I briefly experienced this bug myself, but I couldn't confirm whether it only affected the analyzer for me or also the CFE as VS code ran `dart pub get` automatically and the issue disappeared afterwards on the same code. So I think this might be a Dart bug that we're surfacing, but unfortunately I couldn't find a way to reliably fix this either.

Out of interested, what is the minimum Dart version you declare to support in your pubspec (the `environment.sdk`) field @lance-mlabs @chani22? Does anything break if you upgrade to say `sdk: ^3.4.0` / does that fix the issue?

**Comment by lance-mlabs** [comment]
> Out of interested, what is the minimum Dart version you declare to support in your pubspec (the environment.sdk) field @lance-mlabs @chani22? Does anything break if you upgrade to say sdk: ^3.4.0 / does that fix the issue?

@simolus3 fixing my Dark sdk version (sdk: ^3.4.0) as you suggested has resolved the issue for me! Previously I had `sdk: ">=2.17.0"`. Sounds like a Dart SDK related bug indeed. Thanks for all your help on this!

**Comment by simolus3** [comment]
In that case, fixing this is probably as simple as updating this line https://github.com/simolus3/drift/blob/9155b6856019b3d020032ae451a8082c47a48ed5/drift_dev/lib/src/backends/build/drift_builder.dart#L469

(well, it doesn't actually fix anything but at least we print a visible warning in that case). I'm not sure if 3.4.0 is the actual minimum version or if we should just say 3.0.0.

**Comment by lance-mlabs** [comment]
> I'm not sure if 3.4.0 is the actual minimum version or if we should just say 3.0.0.

On my end, setting `sdk: ^3.0.0` works without issue 👍  So seems reasonable to make that the minimum version.

**Comment by simolus3** [comment]
Fixed in 3898a7b7a5eff00d55b313586c9be0823a7f5e03 - thanks for the report and help investigating this.
